visualization/app/codeCharta/ui/fileExtensionBar/fileExtensionBar.component.ts

```diff
@@ -34,16 +34,13 @@ export class FileExtensionBarController implements CodeMapPreRenderServiceSubscr
 			.getMeshDescription()
 			.buildings.filter(building => building.node.isLeaf)
 
-		const visibleFileExtensions: string[] = this._viewModel.distribution
-			.filter(metric => metric.fileExtension !== "other")
-			.map(metric => metric.fileExtension)
+		const visibleFileExtensions = new Set(
+			this._viewModel.distribution.filter(metric => metric.fileExtension !== "other").map(metric => metric.fileExtension)
```

Comment:
this can be combined again

Reply:
Done. I also combined the buildings for loop with the loop below.